### PR TITLE
feat: Preview by publishing to subdomain only aka staging

### LINKED ISSUE #2644: ⤴️ Preview by publishing to subdomain only aka staging
This feature will allow users to create and test changes in a controlled environment before going live. It’s an ideal solution for reviewing new designs, experimenting with layouts, and testing functionality without affecting the live website.

## Todo

- [ ] allow selecting each domain for publishing
- [ ] only for pro

### PR DESCRIPTION
**Pull request by istarkov**
## Description

closes #2644

- [x] - Merge this https://github.com/webstudio-is/webstudio/pull/4156 into this PR

- If the user has no custom domains:
  - [x] Checkbox for Pro users is not shown.
  - [x] Checkbox for Free users is shown but disabled (Tooltip enabled).

- If the user has custom domains:
  - No Pro:
    - [x] Checkboxes are shown disabled with Tooltips.
  - Pro:
    - Users can select any checkbox (no Tooltips).

- Current indexing logic:
  - If a user has custom domain(s), upon publishing, the `*.wstd.io` domain will receive the meta tag:
    ```html
    <meta name="robots" content="noindex, nofollow">
    ```

- [x] - Fix example.com in tests (NEXT PR #4156) 
- [ ] - saas cleanup should be changed (we need to preserve latest build by domain now) (NEXT PR)
- [x] - Disable Publish Button if no checkboxes are selected

- "With the Pro, you’ll gain the ability to publish to each domain independently!"
image

### Non Pro

- Login as a new non Pro User `pass:vaska@pupkin.ru`
- Create a new Project, Click Publish. Expect  Disabled Checkbox with Tolltip is visible. Publish is Enabled.
image
- Publish, check it published.
- Add custom domain, Publish, check both are published.

### Pro

- Login as a Pro User `pass`
- Open new Project. Expect No checkbox. One wstd domain.
image
- Publish. Expect it published.
- Add domain. See checkboxes visible after domain activated.
- Publish. Expect both published.
- Uncheck checkboxes, see publish button disabled. See it has tooltip.

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env` file

### CODE REVIEW

**Comment by johnsicili** [comment]
It would appear that clicking on the copy button for DNS records refreshes the page:

https://github.com/user-attachments/assets/340ef3f6-d84b-472d-a345-c1514e3b8677

Will continue testing just wanted to note that first

**Review by johnsicili — APPROVED** [review]
1. Would be better UX to disable publishing button if none are checked
2. I see adding the indication that its a Pro feature is left. I would really like to see that.

Otherwise good. Tested on custom domain in staging and works as expected

File: apps/builder/app/builder/features/topbar/add-domain.tsx [hunk 1]
```diff
@@ -94,56 +89,43 @@ export const AddDomain = ({
...
-                isPublishing || сreateState !== "idle" || domainState !== "idle"
-              }
+              disabled={isPending}
               onKeyDown={(event) => {
                 if (event.key === "Enter") {
-                  handleCreate();
+                  buttonRef.current
+                    ?.closest("form")
+                    ?.requestSubmit(buttonRef.current);
                 }
```
Comment (TrySound) [inline apps/builder/app/builder/features/topbar/add-domain.tsx:101]: Don't get it. Button is sibling so form should be an ancestor of event and enter should just work. Why manual handling?

File: apps/builder/app/builder/features/topbar/domain-checkbox.tsx [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+
+  if (project === undefined) {
+    return;
+  }
+
+  const tooltipContentForFreeUsers = hasProPlan ? undefined : (
+    <Flex direction="column" gap="2" css={{ maxWidth: theme.spacing[28] }}>
+      <Text variant="titles">Publish to Staging</Text>
+      <Text>
+        <Flex direction="column">
```
Comment (TrySound) [inline apps/builder/app/builder/features/topbar/domain-checkbox.tsx:35]: why flex with all line breaks?

File: apps/builder/app/builder/features/topbar/add-domain.tsx (hunk 1 above)
Comment (istarkov) [reply apps/builder/app/builder/features/topbar/add-domain.tsx:101]: See formAction we are not submitting default form action

File: apps/builder/app/builder/features/topbar/domain-checkbox.tsx (hunk 2 above)
Comment (istarkov) [reply apps/builder/app/builder/features/topbar/domain-checkbox.tsx:35]: copypasted from the same tooltip, haven’t spent even second in this code

File: apps/builder/app/builder/features/topbar/add-domain.tsx (hunk 1 above)
Comment (TrySound) [reply apps/builder/app/builder/features/topbar/add-domain.tsx:101]: gotcha

**Pull request merged**